src/Sentry.Unity/Integrations/UnityLogHandlerIntegration.cs

```diff
@@ -1,54 +1,102 @@
+    internal sealed class UnityLogHandlerIntegration : ISdkIntegration, ILogHandler
     {
         internal readonly ErrorTimeDebounce ErrorTimeDebounce = new(TimeSpan.FromSeconds(1));
         internal readonly LogTimeDebounce LogTimeDebounce = new(TimeSpan.FromSeconds(1));
         internal readonly WarningTimeDebounce WarningTimeDebounce = new(TimeSpan.FromSeconds(1));
 
-        // TODO: remove 'IEventCapture' in  further iteration
-        private readonly IEventCapture? _eventCapture;
         private readonly IApplication _application;
 
         private IHub? _hub;
         private SentryUnityOptions? _sentryOptions;
 
-        public UnityApplicationLoggingIntegration(IApplication? application = null, IEventCapture? eventCapture = null)
+        private ILogHandler _unityLogHandler = null!; // Set during register
+
+        public UnityLogHandlerIntegration(IApplication? application = null)
         {
             _application = application ?? ApplicationAdapter.Instance;
-            _eventCapture = eventCapture;
         }
 
         public void Register(IHub hub, SentryOptions sentryOptions)
         {
             _hub = hub;
             _sentryOptions = sentryOptions as SentryUnityOptions;
 
-            _application.LogMessageReceived += OnLogMessageReceived;
+            _unityLogHandler = Debug.unityLogger.logHandler;
+            Debug.unityLogger.logHandler = this;
+
             _application.Quitting += OnQuitting;
         }
 
-        // Internal for testability
-        internal void OnLogMessageReceived(string? condition, string? stackTrace, LogType type)
+        public void LogException(Exception exception, UnityEngine.Object context)
+        {
+            CaptureException(exception, context);
+            _unityLogHandler.LogException(exception, context);
+        }
+
+        internal void CaptureException(Exception exception, UnityEngine.Object? context)
         {
-            if (condition is null || _hub?.IsEnabled != true)
+            if (_hub?.IsEnabled is not true)
             {
                 return;
             }
 
-            if (condition.StartsWith(UnityLogger.LogPrefix, StringComparison.Ordinal))
+            // TODO: Capture the context (i.e. grab the name if != null)
+
+            // NOTE: This might not be entirely true, as a user could as well call `Debug.LogException`
+            // and expect a handled exception but it is not possible for us to differentiate
+            // https://docs.sentry.io/platforms/unity/troubleshooting/#unhandled-exceptions---debuglogexception
+            exception.Data[Mechanism.HandledKey] = false;
+            exception.Data[Mechanism.MechanismKey] = "Unity.LogException";
+            _ = _hub.CaptureException(exception);
+
+            // So the next event includes this error as a breadcrumb
+            _hub.AddBreadcrumb(message: exception.GetType() + ": " + exception.Message, category: "unity.logger", level: BreadcrumbLevel.Error);
```

Comment:
Prefer interpolation:

```suggestion
            _hub.AddBreadcrumb(message: $"{exception.GetType()}: {exception.Message}", category: "unity.logger", level: BreadcrumbLevel.Error);
```

Here are some examples how we do this elsewhere:

https://github.com/getsentry/sentry-dotnet/blob/48a78fba69683cc34125d28e20238b27fd9f199d/src/Sentry.Extensions.Logging/SentryLogger.cs#L106-L126

https://github.com/getsentry/sentry-dotnet/blob/48a78fba69683cc34125d28e20238b27fd9f199d/src/Sentry.Serilog/SentrySink.cs#L107-L129